text/2021-09-09-physical-machine-auth.md

```diff
@@ -0,0 +1,189 @@
+# Physical Machine Auth
+
+## Summary
+
+PhysicalMachineChaos-based auth solution, including end-user authorization
+and service-to-service authentication.
+
+## Motivation
+
+Chaos Dashboard's Authentication and Authorization scheme is only applicable
+to single-cluster Kubernetes, and the Chaos Mesh platform needs to be adapted
+to more scenarios, such as multi-cluster Kubernetes, physical machines, cloud
+infrastructure, etc.
+
+### Goals
+
+- end-user authorization on physical machine chaos
+- service-to-service authentication on physical machine chaos
+
+### Non-Goals
+
+- end-user authentication on physical machine chaos (same with other chaos
+which is using token)
+
+## Detailed design
+
+### End-user authorization
+
+#### New Custom Resource: `PhysicalMachine`
+
+Because there is no concept of pod to filter the target physical machine
+when performing chaos injection on a physical machine, a resource needs to
+be introduced to represent the physical machine, which is the new custom
+resource `PhysicalMachine`. Here is a sample of `PhysicalMachine`.
+
+```yaml
+apiVersion: chaos-mesh.org/v1alpha1
+kind: PhysicalMachine
+metadata:
+  name: pm-172.16.112.130
+  namespace: chaos-testing
+  labels:
+    chaos-mesh/physical-machine-group: abc
+    kubernetes.io/arch: arm64
+    kubernetes.io/os: linux
+spec:
+  address: https://172.16.112.130:31767
```

Comment:
Do we need to add the information of TLS in the crd?

Reply:
Do you mean to split the `address` field into `protocol`, `ip` and `port`, which is similar to the `service` or `endpoint` resource?

Reply:
I mean add information like  public key in the crd. I got it, it is generated by Chaos Mesh, so don't need to set it here

Reply:
I mean some information like the public key. I got it, it is generated by Chaos Mesh, so don't need to set it here.